### PR TITLE
Pass CancellationToken to WaitAsync in client

### LINKED ISSUE #12882: Should respect cancellation token when waiting on connection lock.
Waiting on the connection state lock in the `HubConnection` and `HttpConnection` should respect cancellation token

The `WaitConnectionLockAsync` method should take a cancellation token and  pass it into the `WaitAsync` method.
https://github.com/aspnet/AspNetCore/blob/4921dc3aa3496eb3e9fe73255213ce54a0ba29d6/src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs#L1926-L1930

We also  need this in `HttpConnection` here
https://github.com/aspnet/AspNetCore/blob/4921dc3aa3496eb3e9fe73255213ce54a0ba29d6/src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs#L207

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/dotnet/aspnetcore/issues/12882

### CODE REVIEW

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 1]
```diff
@@ -601,7 +601,7 @@ async Task OnStreamCanceled(InvocationRequest irq)
             var readers = default(Dictionary<string, object>);
 
             CheckDisposed();
-            var connectionState = await _state.WaitForActiveConnectionAsync(nameof(StreamAsChannelCoreAsync));
+            var connectionState = await _state.WaitForActiveConnectionAsync(nameof(StreamAsChannelCoreAsync), token: cancellationToken);
```
Comment (halter73) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Can we remove `token = default` from all of our internal methods that wait for the semaphore like WaitForActiveConnectionAsync, SendWithLock, WaitConnectionLockAsync, etc...?

I know WaitConnectionLockAsync in particular has a lot of usages where we don't want to use a cancellation token for close/reconnect logic, but I like being explicit.

Comment (BrennanConroy) [reply src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Fixed

**Review by halter73 — APPROVED** [review]

**Comment by anurse** [comment]
Be aware, branching is underway. I think we can still get this in to preview3 but likely have to retarget.

**Comment by BrennanConroy** [comment]
Switched base and re-running pipeline

**Comment by anurse** [comment]
@dotnet/aspnet-build @Pilchie are we OK to merge to preview3? We'd really like to get this in to preview3 if we can (it's not essential but would be nice to not have to wait a month ;)).

**Comment by Pilchie** [comment]
Go for it.

**Pull request merged**